src/overlays/actors/ovl_En_Ma_Yto/z_en_ma_yto.c

```diff
@@ -21,126 +83,1425 @@ const ActorInit En_Ma_Yto_InitVars = {
+                    return false;
+                }
+            } else if (((this->actor.params & 0x0F00) >> 8) == 0) {
+                return false;
+            }
+            if (gSaveContext.time >= CLOCK_TIME(20, 0) && CURRENT_DAY == 3) {
+                return false;
+            }
+            break;
+
+        case MA_YTO_TYPE_AFTERMILKRUN:
+            // if (!(ProtectedCremia) && !(gSaveContext.weekEventReg[0x34] & 2)) || (PlayedMilkMinigame))
+            if ((!(gSaveContext.weekEventReg[0x34] & 1) && !(gSaveContext.weekEventReg[0x34] & 2)) ||
+                (gSaveContext.weekEventReg[0xE] & 1)) {
+                return false;
+            }
+            break;
+
+        case MA_YTO_TYPE_4:
+            break;
+    }
+
+    return true;
+}
+
+void EnMaYto_InitAnimation(EnMaYto* this, GlobalContext* globalCtx) {
+    switch (this->type) {
+        case MA_YTO_TYPE_DEFAULT:
+            EnMaYto_ChangeAnim(this, 10);
+            break;
+
+        case MA_YTO_TYPE_DINNER:
+            if (CURRENT_DAY == 1) {
+                EnMaYto_ChangeAnim(this, 14);
+            } else {
+                EnMaYto_ChangeAnim(this, 16);
+            }
+            break;
+
+        case MA_YTO_TYPE_BARN:
+            // if (AliensDefeated)
+            if (gSaveContext.weekEventReg[0x16] & 1) {
+                EnMaYto_ChangeAnim(this, 12);
+            } else {
+                EnMaYto_ChangeAnim(this, 8);
+            }
+            break;
+
+        case MA_YTO_TYPE_AFTERMILKRUN:
+            EnMaYto_ChangeAnim(this, 0);
+            break;
+
+        case MA_YTO_TYPE_4:
+            EnMaYto_ChangeAnim(this, 0);
+            break;
+
+        default:
+            EnMaYto_ChangeAnim(this, 0);
+            break;
+    }
+}
+
+void EnMaYto_ChooseAction(EnMaYto* this, GlobalContext* globalCtx) {
+    switch (this->type) {
+        case MA_YTO_TYPE_DEFAULT:
+            EnMaYto_SetupDefaultWait(this);
+            break;
+
+        case MA_YTO_TYPE_DINNER:
+            this->actor.targetMode = 6;
+            EnMaYto_SetupDinnerWait(this);
+            break;
+
+        case MA_YTO_TYPE_BARN:
+            EnMaYto_SetupBarnWait(this);
+            break;
+
+        case MA_YTO_TYPE_AFTERMILKRUN:
+            this->unk_310 = 0;
+            if (INV_CONTENT(ITEM_MASK_ROMANIS) == ITEM_MASK_ROMANIS && (gSaveContext.weekEventReg[0x34] & 1) &&
+                (Rand_Next() & 0x80)) {
+                EnMaYto_SetupBeginWarmFuzzyFeelingCs(this);
+            } else {
+                EnMaYto_SetupAfterMilkRunInit(this);
+            }
+            break;
+
+        case MA_YTO_TYPE_4:
+            this->actor.flags |= 0x10;
+            EnMaYto_SetupWarmFuzzyFeelingCs(this);
+            break;
+
+        default:
+            EnMaYto_SetupDefaultWait(this);
+            break;
+    }
+}
+
+s32 EnMaYto_SearchRomani(EnMaYto* this, GlobalContext* globalCtx) {
+    Actor* actor = globalCtx->actorCtx.actorList[ACTORCAT_NPC].first;
```

Comment:
Would something like npc_actor be helpful with reading the code, rather than it just being actor?
```suggestion
    Actor* npc_actor = globalCtx->actorCtx.actorList[ACTORCAT_NPC].first;
```